common/djangoapps/student/views.py

```diff
@@ -2552,13 +2556,18 @@ def password_reset_confirm_wrapper(request, uidb36=None, token=None):
 
     if request.method == 'POST':
         password = request.POST['new_password1']
-        is_password_valid, password_err_msg = validate_password(user, password)
-        if not is_password_valid:
+        is_security_violated, password_err_msg = validate_password(user, password)
```

Comment:
I still don't like how you reversed the conditional here.
I think the original way makes more sense. The original logic says, 
`if the password is not valid, then regardless whether we have the error message or not, return an error page to user.`
With your new logic, you are basically saying
`If i have error messages, return to user an error page regardless of if the password is valid or not.`